generator/internal/connection_generator.cc

```diff
@@ -207,27 +217,62 @@ class $connection_class_name$ {
  * A factory function to construct an object of type `$connection_class_name$`.
  *
  * The returned connection object should not be used directly; instead it
- * should be passed as an argument to the constructor of $client_class_name$,
- * and that class used instead.
+ * should be passed as an argument to the constructor of $client_class_name$.
  *
- * The optional @p opts argument may be used to configure aspects of the
+ * The optional @p options argument may be used to configure aspects of the
  * returned `$connection_class_name$`. Expected options are any of the types in
  * the following option lists:
  *
  * - `google::cloud::CommonOptionList`
  * - `google::cloud::GrpcOptionList`
  * - `google::cloud::$product_namespace$::$service_name$PolicyOptionList`
  *
- * @note Unrecognized options will be ignored. To debug issues with options set
- *     `GOOGLE_CLOUD_CPP_ENABLE_CLOG=yes` in the environment and unexpected
- *     options will be logged.
- *
+ * @note Unexpected options will be ignored. To log unexpected options instead,
+ *     set `GOOGLE_CLOUD_CPP_ENABLE_CLOG=yes` in the environment.
+ *)""");
+  switch (endpoint_location_style) {
+    case ServiceConfiguration::LOCATION_DEPENDENT:
+    case ServiceConfiguration::LOCATION_DEPENDENT_COMPAT:
+      HeaderPrint(R"""(
+ * @param location Sets the prefix for the default `EndpointOption` value.)""");
+      break;
+    default:
+      break;
+  }
+  HeaderPrint(R"""(
  * @param options (optional) Configure the `$connection_class_name$` created by
  * this function.
  */
+std::shared_ptr<$connection_class_name$> Make$connection_class_name$(
+)""");
+  HeaderPrint("    ");
+  switch (endpoint_location_style) {
+    case ServiceConfiguration::LOCATION_DEPENDENT:
+    case ServiceConfiguration::LOCATION_DEPENDENT_COMPAT:
+      HeaderPrint("std::string const& location, ");
+      break;
+    default:
+      break;
+  }
+  HeaderPrint("Options options = {});\n");
+
+  switch (endpoint_location_style) {
+    case ServiceConfiguration::LOCATION_DEPENDENT_COMPAT:
+      HeaderPrint(R"""(
+/**
+ * A backwards-compatible version of the previous factory function. The
+ * default value of the `EndpointOption` is useless in this case, and so
```

Comment:
I am not sure it is always useless, some services have global and regional resources, with global and regional endpoints.  Maybe "Unless the service also offers a global endpoint, the default value of the `EndpointOption` may be useless ..." ?